pineappl/src/grid.rs

```diff
@@ -385,6 +388,48 @@ pub struct EkoInfo {
     pub lumi_id_types: String,
 }
 
+/// Information about the evolution kernel operator (EKO) passed to [`Grid::evolve`] as `operator`,
+/// which is used to convert a [`Grid`] into an [`FkTable`]. The dimensions of the EKO must
+/// correspond to the values given in [`fac1`], [`pids0`], [`x0`], [`pids1`] and [`x1`], exactly in
+/// this order. Members with a `1` are defined at the squared factorization scales given in
+/// [`fac1`] (often called process scales) and are found in the [`Grid`] that [`Grid::evolve`] is
+/// called with. Members with a `0` are defined at the squared factorization scale [`fac0`] (often
+/// called fitting scale or starting scale) and are found in the [`FkTable`] resulting from
+/// [`Grid::evolve`].
+///
+/// The EKO may convert a `Grid` from a basis given by the particle identifiers [`pids1`] to a
+/// possibly different basis given by [`pids0`]. This basis must also be identified using
+/// [`lumi_id_types`], which tells [`FkTable::convolute`] how to perform a convolution. The members
+/// [`ren1`] and [`alphas`] must be the strong couplings given at the respective renormalization
+/// scales. Finally, [`xir`] and [`xif`] can be used to vary the renormalization and factorization
+/// scales, respectively, around their central values.
+pub struct OperatorInfo {
+    /// Squared factorization scales of the `Grid`.
+    pub fac1: Vec<f64>,
+    /// Particle identifiers of the `FkTable`.
+    pub pids0: Vec<i32>,
+    /// `x`-grid coordinates of the `FkTable`
+    pub x0: Vec<f64>,
+    /// Particle identifiers of the `Grid`. If the `Grid` contains more particle identifiers than
+    /// given here, the contributions of them are silently ignored.
+    pub pids1: Vec<i32>,
+    /// `x`-grid coordinates of the `Grid`.
+    pub x1: Vec<f64>,
+
+    /// Squared factorization scale of the `FkTable`.
+    pub fac0: f64,
+    /// Renormalization scales of the `Grid`.
+    pub ren1: Vec<f64>,
```

Comment:
Maybe you can re-sort 0 and 1 variables, to make a `Grid` section and an `FkTable` section, something like

```
x0
pids0
ren0
fac0
x1
pids1
ren1
fac1
```

It is not fundamental, but:
> Readability counts.

Reply:
We can do that, but my idea was to have the same ordering as in the operator, then all other information. What do you think?

Reply:
:thinking: maybe I'm not understanding what you mean by "operator".

The EKO object should not be sorted, and if it is might be that is rather at random.

Reply:
What do you mean by *operator indices*? The internal sorting of each grid? I'm not understanding...

Reply:
as in `[q2][pid0][x0][pid1][x1]`

Reply:
Now I'm actually wondering what difference the ordering of members in Rust has; if I want to initialize them I can apparently choose any order ...

Reply:
I reordered the members in commit fec9c0c16fb409a70c50cfb5417747e151fe8143. Incidentally this is the order which PineAPPL would like best.